File: flask/src/controller/auth_controller.py
from flask import request
from service.auth_service import login, logout
from response.response import response
from response.response_error import ResponseError

# Controller untuk menangani proses login pengguna
def login_controller():
    try:
        data = request.get_json()
        result = login(data)
        return response(200, result, "Login success")
    except ResponseError as e:
        raise e
    
# Controller untuk menangani proses logout pengguna
def logout_controller():
    try:
        token = request.headers.get("Authorization")
        result = logout(token)
        return response(200, result, "Logout success")
    except ResponseError as e:
        raise e

File: express/src/controller/auth-controller.js
import authService from "../service/auth-service.js";
import { response } from "../response/response.js";

// Controller untuk login user
const login = async (req, res, next) => {
  try {
    const result = await authService.login(req.body);
    response(200, result, "Login success", res);
  } catch (error) {
    next(error);
  }
};

// Controller untuk logout user
const logout = async (req, res, next) => {
  try {
    const token = req.get("Authorization");
    const result = await authService.logout(token);
    response(200, result, "Logout success", res);
  } catch (error) {
    next(error);
  }
};

export default {
  login,
  logout,
};
